migrationTool/contracts/Manifest.cs

```diff
@@ -51,9 +51,9 @@ public Track(StreamType type)
         // Check if the track is stored as one file per fragment.
         public bool IsMultiFile => string.IsNullOrEmpty(Path.GetExtension(Source));
 
-        public uint TrackID => uint.Parse(Parameters.Single(p => p.Name == "trackID").Value);
+        public uint TrackID => uint.Parse(Parameters?.SingleOrDefault(p => p.Name == "trackID")?.Value ?? "1");
```

Comment:
Parameters should never be null (Initialized to empty array) so the first ? is not needed. But parameters themselves are optional so the second one is needed.

Reply:
You're right, Parameters should not be null because they have a default initializer, but they can be assigned to null using the setter. I just wanted to make sure to catch this also.